### PR TITLE
API layer changes to support preferred notification channels

### PR DESCRIPTION
**Pull request by acn-sbuad**
## Description

API layer changes to support preferred notification channels

## Related Issue(s)
- #541

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [x] Manual testing done (required)
- [x] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [x] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

File: src/Altinn.Notifications/Controllers/OrdersController.cs [hunk 1]
```diff
@@ -24,14 +29,18 @@ namespace Altinn.Notifications.Controllers;
...
 {
+    private readonly IValidator<NotificationOrderRequestExt> _validator;
     private readonly IGetOrderService _getOrderService;
+    private readonly IOrderRequestService _orderRequestService;
 
     /// <summary>
     /// Initializes a new instance of the <see cref="OrdersController"/> class.
     /// </summary>
-    public OrdersController(IGetOrderService getOrderService)
+    public OrdersController(IValidator<NotificationOrderRequestExt> validator, IGetOrderService getOrderService, IOrderRequestService orderRequestService)
```
Comment (acn-sbuad) [inline src/Altinn.Notifications/Controllers/OrdersController.cs:39]: sonarcloud complains about multiple resopnsibilities for a single controller. Do we want to split it into OrdersQueryController and OrdersCommandController or keep it as is?

File: src/Altinn.Notifications/Validators/Rules/RecipientRules.cs [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+                .WithMessage("Organization number cannot be combined with email address, mobile number, or national identity number.");
+        });
+    }
+
+    /// <summary>
+    /// Validated as email address based on the Altinn 2 regex
+    /// </summary>
+    /// <param name="email">The string to validate as an email address</param>
+    /// <returns>A boolean indicating that the email is valid or not</returns>
+    internal static bool IsValidEmail(string? email)
```
Comment (HenningNormann) [inline src/Altinn.Notifications/Validators/Rules/RecipientRules.cs:230]: Doesn't the built-in validator for email do the job?

File: src/Altinn.Notifications/Controllers/OrdersController.cs (hunk 1 above)
Comment (HenningNormann) [reply src/Altinn.Notifications/Controllers/OrdersController.cs:38]: I suggest keep

File: src/Altinn.Notifications/Validators/Rules/RecipientRules.cs (hunk 2 above)
Comment (acn-sbuad) [reply src/Altinn.Notifications/Validators/Rules/RecipientRules.cs:230]: Our regex is a bit more robust than the built-in validator which allowed the following test cases which we expect to fail: user.@example.com, user@exa!mple.com and user@example..com

**Review by HenningNormann — APPROVED** [review]

**Pull request merged**